services/src/main/scala/cromwell/services/healthmonitor/HealthMonitorServiceActor.scala

```diff
@@ -0,0 +1,119 @@
+package cromwell.services.healthmonitor
+
+import java.util.concurrent.TimeoutException
+
+import akka.actor.{Actor, Scheduler, Timers}
+import akka.pattern.{after, pipe}
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+import scala.language.postfixOps
+import HealthMonitorServiceActor._
+import com.typesafe.config.Config
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+import net.ceedubs.ficus.Ficus._
+
+trait HealthMonitorServiceActor extends Actor with LazyLogging with Timers {
+  val serviceConfig: Config
+  def subsystems: List[MonitoredSubsystem]
+
+  implicit val ec: ExecutionContext = context.system.dispatcher
+
+  lazy val futureTimeout: FiniteDuration = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.check-timeout").getOrElse(DefaultFutureTimeout)
+  lazy val staleThreshold: FiniteDuration = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.status-ttl").getOrElse(DefaultStaleThreshold)
```

Comment:
aren't these constants for all implementors that are getting reevaluated for each implementor? i.e. shouldn't this move to the companion object

Reply:
Yeah I think I was thinking this had to happen in the class, but you're right

Reply:
Ah, I see why I did this. They depend on `serviceConfig` which is a class parameter

Reply:
@mcovarr let me know if you find my "Ah, i see why i did this" explanation sufficient. I could still have something in the companion object which takes the serviceConfig, that just seemed not worth it to me.

Reply:
yeah I missed that point before, this is fine.  seagulling withdrawn.